src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts

```diff
@@ -208,8 +208,15 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
             ${network.routers.map((router) => {
               const otbr =
                 this._otbrInfo && this._otbrInfo[router.extended_address];
-              const showOverflow =
-                ("dataset" in network && router.border_agent_id) || otbr;
+              const showDefaultRouter = !!network.dataset;
+              const isDefaultRouter =
+                showDefaultRouter &&
+                ((network.dataset!.preferred_border_agent_id &&
+                  router.border_agent_id ===
+                    network.dataset!.preferred_border_agent_id) ||
```

Comment:
Why did we add border agent id to this?

Reply:
Hm, yeah not sure. I think I had a probably old datastore which had the preferred border agent id set but not the preferred extended address :thinking: . But today we have to rely more on the extended address since Apple still does not have a border agent id :cry: .

Reply:
I'll drop the border_agent_id check.